### PR TITLE
fix: internationalization and slug text domain congruence

### PR DESCRIPTION
**Pull request by MaferMazu**
## Description

This PR:
- Removes the `phpcs:ignore WordPress.WP.I18n.NonSingularStringLiteralContext`
- Changes the text domain used in the internationalization to be congruent with the text domain in the `openedx-commerce.php` header.
- Add a test to check that the text domain is congruent.
- I also excluded composer-setup.php from the phpcs. That is a local file that does not follow the phpcs WordPress standards.

## Testing instructions

- Check if you can still create an Open edX Enrollment Request.
- Check the tests.

## Additional information

I added woocommerce as part of the rules because we added some text in the woocommerce pages.
This change is part of the work to publish the plugin in the WordPress Plugin Directory.

> Internationalization: Don't use variables or defines as text, context or text domain parameters.
> 
> Please do not use variable names for the text, context or text domain portion of a gettext function, all of them NEED to be strings, otherwise they could not be read by the translation parser.
> 
> For example, this is a bad call: esc_html__( 'Translate me.' , $plugin_slug ); , the fix here would be like this esc_html__( 'Translate me.' , 'plugin-slug' );
> 
> This is also wrong: esc_html__( 'Your city is ' . $city . '.' , 'plugin-slug' ); , the fix here would be using a placeholder and change it later using printf , like this:
> print(
> 
>  /* translators: %s: Name of a city */
> 
>  esc_html__( 'Your city is %s.', 'plugin-slug' ),
> 
>  esc_html($city)
> 
> );
> 
> Also have in mind that your text domain has to be the same as your plugin's slug (also known as permalink).
> 
> The text domain also needs to be added to the plugin header. WordPress uses it to internationalize your plugin meta-data even when the plugin is disabled.
> 
> You can read https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#text-domains for more information.
> 
> Example(s) from your plugin:
> openedx-commerce/includes/model/class-openedx-commerce-post-type.php:104 _x('Add New', $this->post_type, 'wp-openedx-commerce');

## References
WordPress text domains doc: https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#text-domains
Example of phpcs.xml file for WordPress Text Domain: https://github.com/WordPress/WordPress-Coding-Standards/blob/develop/phpcs.xml.dist.sample#L105

## Checklist for Merge

- [x] Tested in a remote environment
- [x] Rebased master/main
- [ ] Squashed commits

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @MaferMazu! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Review by felipemontoya — APPROVED** [review]
Thanks @MaferMazu, this PR looks great.

**Comment by felipemontoya** [comment]
@MaferMazu now that I'm a proper CC for this repo, my review will unlock the merge button. I'll let you merge at your convenience

**Comment by MaferMazu** [comment]
I will merge this knowing that the release workflow is going to fail, but expecting merge #71 soon to fix the issue.

**Comment by openedx-webhooks** [comment]
@MaferMazu 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-wordpress-ecommerce&entry.867055334=https://github.com/openedx/openedx-wordpress-ecommerce/pull/69&entry.1484655318=https://github.com/MaferMazu&entry.752974735=2024-03-08+01:53&entry.1917517419=2024-04-02+20:30&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
